interview-questions.md

```diff
@@ -7,44 +7,69 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating an API with a model called Animal that has_many Sightings, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be part of the Animal model or the Sightings model?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: In order to fix the mistake you need to add a column to the Sightings model. You would do $rails g migration add_foreign_key_to_sightings in the terminal. Then in the migration file in def change we would type add_column :sightings, :animal_id, :integer. Then you would do $rails db:migrate in the terminal so it can be added to the schema. The name of the foreign key is animal_id and it would be part of the Sightings model.
 
+  Researched answer: In my research it looked like I was close but not quite there. I do need to $rails g migration add_foreign_key_to_sightings but then inside the file under def change I would type add_foreign_key :sightings, :animal because sightings belongs to animal. Then $rails db:migrate to add to the schema.
 
 
 2. Which RESTful API routes must always be passed params? Why?
 
-  Your answer:
+  Your answer: The RESTful ApI routes that need params are GET when referring to show because it needs to know specifically which item to show the user. PUT needs a parameter when updating or editing because the program needs to know which item it is updating and finally DELETE requires a parameter so the program knows which item needs to be removed.
 
-  Researched answer:
+  Researched answer: The RESTful API routes are GET, PUT, POST AND DELETE. GET  retrieves a source. If you want a specific source then a param needs to be passed. PUT changes the state or updates a source, which in order for that to be accomplished a param is needed here as well. POST creates a resource so a param is not needed because a param does not exist. DELETE removes a source and needs a param so the program knows which source is to be removed.
```

Comment:
You are super close on this! Luckily I am reviewing it while we are going over the differences between restful routes and HTTP verbs. Everything thing here is correct except for the layer of the command that you are requiring. i.e. show, not GET

Reply:
Yep, took notes on the differences for the future.